src/OidcClient.php

```diff
@@ -130,6 +136,21 @@ public function generateAuthorizationRedirect(
       $data['prompt'] = $prompt;
     }
 
+    try {
+      // Check if code challenge is activated
+      if (null !== $this->codeChallengeMethod) {
+        $codeVerifier  = $this->generateCodeVerifier();
+        $codeChallenge = $this->generateCodeChallenge($codeVerifier);
+
+        $data = array_merge($data, [
+            'code_challenge'        => $codeChallenge,
+            'code_challenge_method' => $this->codeChallengeMethod,
+        ]);
+      }
+    } catch (\Exception $e) {
+      throw new OidcAuthenticationException('An error occurred during code challenge generation', $e);
```

Comment:
```suggestion
      throw new OidcException('An error occurred during code challenge generation', $e);
```

The authentication exception is for use within Symfony authenticator, this is not part of that imo